scipy/linalg/tests/test_basic.py

```diff
@@ -931,40 +930,121 @@ def test_check_finite(self):
 
 class TestDet:
     def setup_method(self):
-        np.random.seed(1234)
-
-    def test_simple(self):
-        a = [[1, 2], [3, 4]]
-        a_det = det(a)
-        assert_almost_equal(a_det, -2.0)
-
-    def test_simple_complex(self):
-        a = [[1, 2], [3, 4j]]
-        a_det = det(a)
-        assert_almost_equal(a_det, -6+4j)
-
-    def test_random(self):
-        basic_det = linalg.det
-        n = 20
-        for i in range(4):
-            a = random([n, n])
-            d1 = det(a)
-            d2 = basic_det(a)
-            assert_almost_equal(d1, d2)
-
-    def test_random_complex(self):
-        basic_det = linalg.det
-        n = 20
-        for i in range(4):
-            a = random([n, n]) + 2j*random([n, n])
-            d1 = det(a)
-            d2 = basic_det(a)
-            assert_allclose(d1, d2, rtol=1e-13)
-
-    def test_check_finite(self):
-        a = [[1, 2], [3, 4]]
-        a_det = det(a, check_finite=False)
-        assert_almost_equal(a_det, -2.0)
+        self.rng = np.random.default_rng(1680305949878959)
+
+    def test_1by1_stacked_input_output(self):
+        a = self.rng.random([4, 5, 1, 1], dtype=np.float32)
+        deta = det(a)
+        assert deta.dtype.char == 'd'
+        assert deta.shape == (4, 5)
+        assert_almost_equal(deta, np.squeeze(a))
+
+        a = self.rng.random([4, 5, 1, 1], dtype=np.float32)*np.complex64(1.j)
+        deta = det(a)
+        assert deta.dtype.char == 'D'
+        assert deta.shape == (4, 5)
+        assert_almost_equal(deta, np.squeeze(a))
```

Comment:
per numpy's recommendation
![image](https://user-images.githubusercontent.com/6570539/230786954-f550664f-b9c2-4dfe-9834-65f6360e4469.png)
```suggestion
        assert_allclose(deta, np.squeeze(a))

        a = self.rng.random([4, 5, 1, 1], dtype=np.float32)*np.complex64(1.j)
        deta = det(a)
        assert deta.dtype.char == 'D'
        assert deta.shape == (4, 5)
        assert_allclose(deta, np.squeeze(a))
```
(Here and throughout)

Reply:
Ha I corrected and then wronged these. My bad.